symphony-bdk-legacy/symphony-api-client-java/src/test/java/utils/DataProviderTest.java

```diff
@@ -0,0 +1,110 @@
+package utils;
+
+import clients.SymBotClient;
+import clients.symphony.api.UsersClient;
+import configuration.SymConfig;
+import lombok.SneakyThrows;
+import model.UserInfo;
+import org.junit.Before;
+import org.junit.Test;
+import org.symphonyoss.symphony.messageml.exceptions.InvalidInputException;
+import org.symphonyoss.symphony.messageml.util.IUserPresentation;
+
+import javax.ws.rs.core.NoContentException;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class DataProviderTest {
+  private SymBotClient symBotClient;
+  private SymConfig config;
+
+  @Before
+  public void initClient() {
+    symBotClient = mock(SymBotClient.class);
+    config = mock(SymConfig.class);
+    when(symBotClient.getConfig()).thenReturn(config);
+  }
+
+  @Test
+  public void getUserPresentationByEmailTest() {
+    when(config.getSupportedUriSchemes()).thenReturn(new ArrayList<>());
+    DataProvider dataProvider = new DataProvider(symBotClient);
+    dataProvider.setUserPresentation(12345L, "Test User", "testUser", "testuser@email.com");
+    try {
+      IUserPresentation user = dataProvider.getUserPresentation("testuser@email.com");
+      assertEquals(user.getId(), 12345L);
+      assertEquals(user.getScreenName(), "Test User");
+      assertEquals(user.getPrettyName(), "testUser");
+      assertEquals(user.getEmail(), "testuser@email.com");
+    } catch (InvalidInputException e) {
+      fail();
+    }
```

Comment:
`try/catch` block is not necessary if your test method `throws Exception`. It will make the code easier to read